### DISCUSSION TITLE
Implement pure ESP-IDF code base as a long term goal?

### DISCUSSION THREAD

**Discussion by OOHehir** [LIFsCode/ELOC-3.0#106] [Ideas]
@EDsteve @LIFsCode 

As we know there is certain constraints/ issues about using the Arduino version of the ESP code base, for example, with the new version of the I2S API. The Arduino ESP-IDF has recently being bumped to v4.4.7 ([link](https://github.com/espressif/arduino-esp32/releases)) while the ESP-IDF is now at v5.5 ([link](https://github.com/espressif/esp-idf/releases)) , so a significant lag behind. There would probably be significant improvements & fixes to move to a more up-to-date code base.

The reason for using the Arduino version is probably to facilitate the use of some of the libraries, the main one being ArduinoJSON. It appears that this could be compatible with [ESP-IDF](https://components.espressif.com/components/bblanchon/arduinojson) so perhaps at some point thought should be given to moving to pure ESP-IDF code?

This is a long term goal, but perhaps one to consider?

**Comment by OOHehir** [comment]
@EDsteve @LIFsCode I've just tried to change the line in platform.ini to ```platform = espressif32@~6.9.0``` & got some strange build errors. Could be an issue on my setup but during investigation I came across this [post](https://github.com/platformio/platform-espressif32/issues/1225#issuecomment-2560144650). It seems that keeping the project up-to-date with PlatofromIO/Arduino/Espressif is going to be challenging..

**Comment by EDsteve** [comment]
Yes. Moving all to ESP-IDF was planned as long term goal.
But yeah. I also just read through the post you linked. Doesn't look so good. But things can change. Hopefully :/

**Comment by EDsteve** [comment]
Reading your linked forum post a bit further. They mention pioarduino. Would that be a solution?

**Comment by OOHehir** [comment]
@EDsteve Perhaps it might be, but I think it's just a fork of PlatformIO? If Espressif are not going to support the PlatformIO/ Arduino/ Espressif combination it means relying on a community effort in pioarduino to support it.
